Process_of_Authoring_GitHub_Actions_for_Azure.md

```diff
@@ -0,0 +1,98 @@
+
+## Guidelines for setting permissions on the repo
+Once the repo is created, following is the permission model that is followed across various action repos within the [**Azure**](https://github.com/Azure) and [**Microsoft**](https://github.com/Microsoft) GitHub orgs:
+
+- [actions-admin](https://github.com/orgs/Azure/teams/actions-admin/members) team would be added with "admin" role to every new GH repo created for authoring any new action under the Azure org (github.com/azure/). This will ensure that ACE team can act in admin role to ensure that the process guidelines for authoring & publishing the actions are completely followed.
+
+- Owners of the individual actions could manage the "maintainer" permissions for their teams by [creating a new github team handle](https://github.com/orgs/Azure/new-team) for their crews and adding this handle with "admin"/"maintainer" role as needed. 
+
+**Note:** It is recommended that team handles are used for maintaining permissions instead of adding individual member's github handles.
+
+- Kindly request the corresponding Azure product teams to include their team handle as "Admins"/"Maintainers" to facilitate raising any commits from their side for feature enhancements or for fixing supportability issues
+
+
+## ReadMe.md &amp; action.yml files
+
+Once the Action development is complete, ensure that it is documented well in the **readme.md** file at the root of the repo of the Master branch. Please note that this content is used when the Action is published to the GitHub Marketplace. Hence including a sample workflow which covers the best usage of the Action is always recommended.
+
+Also ensure that the **action.yml** is present in the root of the repo and has the Action name &amp; description well defined. Also choose the icon &amp; color suitably under branding as these are used when the Action is listed in Marketplace catalog.
+
+Note: All actions published from **Azure** GitHub org are branded using the Azure icon irrespective of the icon used in the action.yml. This is to leverage the recall value of Azure brand and icon as compared to the individual icons of various Azure services. 
+
+## Action Versioning
+
+**Versioning Guidance for GH actions that should be adhered to before a new action or an enhancement to an existing action is released:** 
+
+Official Release Management Guidance from GH platform team: https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/about-actions#using-release-management-for-actions
+
+**Note:** Introduce a new major version of the branch (releases/v2) and tag (v2) if changes break the existing workflows using the previous version. In case of Minor version releases, ensure to move the major version tag to always point to the most recent minor version tag to ensure all the existing users of the major version would automatically gain the benefits from the new enhancements added in the minor version release.
+
+
+## Publish the action to Marketplace
```

Comment:
Should we think about creating issue template for publishing. The recommendation would be for creating issue before release and ensure all the items listed in the template are handled.

Reply:
This is a good idea. However lets be sure about who would finally approve and publish the release - be it major or minor path update to the Action. IMO ACE could be involved for the first major release but for all the further updates (major & minor), we could just be informed as FYI than be the bottlenecks as it might also create an overhead for ACE. Thoughts?

Reply:
We can have a new item for tracking issue template for publishing

Reply:
https://github.com/github/GitHub-EcoSystem/issues/1103